packages/cursorless-engine/src/processTargets/targets/BoundedParagraphTarget.ts

```diff
@@ -0,0 +1,135 @@
+import { Range } from "@cursorless/common";
+import { BaseTarget, MinimumTargetParameters } from "./BaseTarget";
+import { LineTarget } from "./LineTarget";
+import { expandToFullLine } from "../../util/rangeUtils";
+import { InteriorTarget, ParagraphTarget } from ".";
+
+interface BoundedParagraphTargetParameters extends MinimumTargetParameters {
+  readonly paragraphTarget: ParagraphTarget;
+  readonly containingInterior: InteriorTarget;
+}
+
+export class BoundedParagraphTarget extends BaseTarget<BoundedParagraphTargetParameters> {
+  readonly type = "BoundedParagraphTarget";
+  readonly insertionDelimiter = "\n\n";
+  readonly isLine = true;
+  private containingInterior: InteriorTarget;
+  private paragraphTarget: ParagraphTarget;
+  private startLineGap: number;
+  private endLineGap: number;
+
+  constructor(parameters: BoundedParagraphTargetParameters) {
+    super({
+      ...parameters,
+      contentRange: getIntersectionStrict(
+        parameters.paragraphTarget.contentRange,
+        parameters.containingInterior.contentRange,
+      ),
+    });
+
+    this.containingInterior = parameters.containingInterior;
+    this.paragraphTarget = parameters.paragraphTarget;
+    this.startLineGap =
+      this.contentRange.start.line -
+      this.containingInterior.fullInteriorRange.start.line;
+    this.endLineGap =
+      this.containingInterior.fullInteriorRange.end.line -
+      this.contentRange.end.line;
+  }
+
+  getLeadingDelimiterTarget() {
+    return this.startLineGap > 1
+      ? this.paragraphTarget.getLeadingDelimiterTarget()
+      : undefined;
+  }
+
+  getTrailingDelimiterTarget() {
+    return this.endLineGap > 1
+      ? this.paragraphTarget.getTrailingDelimiterTarget()
+      : undefined;
+  }
+
+  getRemovalRange(): Range {
```

Comment:
Copied from `ParagraphTarget` with addition of `startLineGap` check in the middle